go/vt/vtadmin/api.go

```diff
@@ -125,6 +127,81 @@ func (api *API) ListenAndServe() error {
 	return api.serv.ListenAndServe()
 }
 
+// FindSchema is part of the vtadminpb.VTAdminServer interface.
+func (api *API) FindSchema(ctx context.Context, req *vtadminpb.FindSchemaRequest) (*vtadminpb.Schema, error) {
+	span, _ := trace.NewSpan(ctx, "API.FindSchema")
+	defer span.Finish()
+
+	span.Annotate("table", req.Table)
+
+	clusters, _ := api.getClustersForRequest(req.ClusterIds)
+
+	var (
+		m       sync.Mutex
+		wg      sync.WaitGroup
+		rec     concurrency.AllErrorRecorder
+		results []*vtadminpb.Schema
+	)
+
+	for _, c := range clusters {
+		wg.Add(1)
+
+		go func(c *cluster.Cluster) {
+			defer wg.Done()
+
+			tablets, err := c.FindTablets(ctx, func(t *vtadminpb.Tablet) bool {
+				// Filter out all the non-serving tablets once, to make the
+				// later, per-keyspace filtering slightly faster (fewer
+				// potentially-redundant iterations).
+				return t.State == vtadminpb.Tablet_SERVING
+			}, -1)
+			if err != nil {
+				err := fmt.Errorf("could not find any serving tablets for cluster %s: %w", c.ID, err)
+				rec.RecordError(err)
+
+				return
+			}
+
+			schemas, err := api.getSchemas(ctx, c, tablets)
+			if err != nil {
+				err := fmt.Errorf("%w: while collecting schemas for cluster %s", err, c.ID)
+				rec.RecordError(err)
+
+				return
+			}
+
+			for _, schema := range schemas {
+				for _, td := range schema.TableDefinitions {
+					if td.Name == req.Table {
+						m.Lock()
+						results = append(results, schema)
+						m.Unlock()
+
+						return
+					}
+				}
+			}
+
+			log.Infof("cluster %s has no tables named %s", c.ID, req.Table)
+		}(c)
+	}
+
+	wg.Wait()
+
+	if rec.HasErrors() {
+		return nil, rec.Error()
+	}
+
+	switch len(results) {
+	case 0:
+		return nil, fmt.Errorf("%w: no schemas found with table named %s", errors.ErrNoSchema, req.Table)
+	case 1:
+		return results[0], nil
+	default:
+		return nil, fmt.Errorf("%w: %d schemas found with table named %s", errors.ErrAmbiguousSchema, len(results), req.Table)
```

Comment:
Is it convenient for the API to return the matches, so the front-end can show something like "did you mean..." with links? 

The front-end will _likely_ have all the schemas fetched anyway (since at least one fetch is required to populate the sidebar counts) so it's not strictly necessary, but having them as part of the "_n_ schemas found" response would be ideal (and would feel a lot smoother).

Reply:
yeah, we can definitely do that!

Reply:
ah, hm, i responded too quickly. given that the grpc method returns a single `Schema`, we'll need to pack this info into the error somehow .... can we defer doing this until we know how the frontend is going to want to consume it?

Reply:
Oh, ha. Yeah, that sounds cumbersome. We can definitely revisit it later. We might not even need to.

Reply:
https://github.com/vitessio/vitess/issues/7611